app/code/Magento/WishlistGraphQl/Model/AddItemsToWishlist.php

```diff
@@ -0,0 +1,211 @@
+        }
+
+        if (count($errors) > 0) {
+            throw new GraphQlInputException(__(implode("\n", $errors)));
+        }
+
+        return $wishlist;
+    }
+
+    /**
+     * Creates buy request
+     *
+     * @param array $item
+     * @return DataObject
+     * @throws GraphQlInputException
+     */
+    private function createBuyRequest(array $item): DataObject
+    {
+        $cartItemData['data'] = [
+            'quantity' => $item['quantity'] ?? 1,
+            'sku' => $item['sku']
+        ];
+
+        if (isset($item['parent_sku'])) {
+            $cartItemData['parent_sku'] = $item['parent_sku'];
+        }
+
+        $cartItemData = array_merge_recursive(
+            $cartItemData,
+            $this->processEnteredOptions($item),
+            $this->processSelectedOptions($item)
+        );
+
+        return $this->buyRequestBuilder->build($cartItemData);
+    }
+
+    /**
+     * Extracts entered options for item
+     *
+     * @param array $item
+     * @return array
+     * @throws GraphQlInputException
+     */
+    private function processEnteredOptions(array $item): array
+    {
+        $enteredOptions = [];
+
+        if (!isset($item['entered_options'])) {
+            return $enteredOptions;
+        }
+
+        foreach ($item['entered_options'] as $enteredOption) {
+            $option = $this->getProductOptionFromRequest->execute(
+                $enteredOption,
+                ProductOptionInterface::SOURCE_ENTERED
+            );
+
+            if ($option->getType() === ProductOptionInterface::TYPE_CUSTOM) {
+                $enteredOptions['customizable_options'][] = [
+                    'id' => $option->getId(),
+                    'value_string' => $option->getValue()
+                ];
+            } else {
+                throw new GraphQlInputException(__("Invalid option type '{$option->getType()}'"));
+            }
+        }
+
+        return $enteredOptions;
+    }
+
+    /**
+     * Extracts selected options for item
+     *
+     * @param array $item
+     * @return array
+     * @throws GraphQlInputException
+     */
+    private function processSelectedOptions(array $item): array
+    {
+        $selectedOptions = [];
+
+        if (!isset($item['selected_options'])) {
+            return $selectedOptions;
+        }
+
+        foreach ($item['selected_options'] as $selectedOption) {
+            $option = $this->getProductOptionFromRequest->execute(
+                ['value' => $selectedOption],
+                ProductOptionInterface::SOURCE_SELECTED
+            );
+            if ($option->getType() === ProductOptionInterface::TYPE_CUSTOM) {
+                $selectedOptions['customizable_options'][] = [
+                    'id' => $option->getId(),
+                    'value_string' => $option->getValue()
+                ];
+            } elseif ($option->getType() === ProductOptionInterface::TYPE_CONFIGURABLE) {
```

Comment:
Mayne makes sense to split logic into different modules
The base module will provide only an extension point (Strategy pattern)
See example

https://github.com/magento/async-import/blob/2.3-asynchronous-import-module/app/code/Magento/AsynchronousImportDataExchangingApi/Model/ExchangeDataStrategyInterface.php

https://github.com/magento/async-import/blob/2.3-asynchronous-import-module/app/code/Magento/AsynchronousImportDataExchanging/Model/ExchangeImportData.php#L94
But instead of using ObjectManager directly need to use Magento `Proxy` (in DI, NOT in constructor)